allure-cucumber4-jvm/src/main/java/cucumber/runtime/formatter/TestSourcesModelProxy.java

```diff
@@ -0,0 +1,33 @@
+package cucumber.runtime.formatter;
+
+import cucumber.api.event.TestSourceRead;
+import gherkin.ast.Feature;
+import gherkin.ast.ScenarioDefinition;
+
+/**
+ * Proxy class to internal Cucumber implementation of TestSourcesModel.
+ */
+public class TestSourcesModelProxy {
+
+    private final TestSourcesModel testSources;
+
+    public TestSourcesModelProxy() {
+        this.testSources = new TestSourcesModel();
```

Comment:
Did we determine if this can be called concurrently? 
If so internally this class contains HashMaps of the parsed Feature file data which could cause problems if its updated from multiple threads. If so we could wrap this in a thread-local too?

NB. I didn't encounter any issues with he code when using for the past month which lead me to believe the plugin is called sequentially, but would need to check the cucumber code.

Reply:
> NB. I didn't encounter any issues with he code when using for the past month which lead me to believe the plugin is called sequentially, but would need to check the cucumber code.

I'd like to have our code thread safe any way in order to prevent issues if cucumber team decide to make internal changes and run code in parallel. But we can target this change to future releases

Reply:
I was looking at existing formatters, and that's how they use it in built-in formatter, including those that implement ConcurrentEventListener - e.g. [TimelineFormatter](https://github.com/cucumber/cucumber-jvm/blob/c3d167f0109d29b08ad571f279f3ca65afae6ae3/core/src/main/java/cucumber/runtime/formatter/TimelineFormatter.java)